python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -621,6 +619,33 @@ def evaluate(self, *, cache: MutableMapping[int, DataFrame]) -> DataFrame:
         results = [
             req.evaluate(result_subs, mapping=mapping) for req in self.agg_requests
         ]
+        # Handle order preservation of groups
+        # like cudf classic does
+        # https://github.com/rapidsai/cudf/blob/5780c4d8fb5afac2e04988a2ff5531f94c22d3a3/python/cudf/cudf/core/groupby/groupby.py#L723-L743
+        if self.maintain_order:
```

Comment:
You need the same broadcast call for the results before doing this dance I think

Reply:
I think I've moved it up like you wanted.

I don't know how to test this, though.
Can you recommend a test case?

Reply:
Should happen if you write one of the `groupby` keys as a `lit` (for example `df.group_by("a", pl.lit(1)).agg(...)`

Reply:
Updated.